### PR TITLE
UI: Added expand all column button and full screen view for lineage

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on the UI: Added expand all column button and full screen view for lineage
#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/71748675/198675004-7f373617-7769-45c6-b066-1bb3bfbe2eeb.mov

updated screenshot

image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by ShaileshParmar11** [comment]
Lineage is still in progress feature, unit test will be added in last PR

File: openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+              entityLineage={entityLineage}
+              entityLineageHandler={entityLineageHandler}
+              entityType={entityType}
+              isLoading={isLineageLoading}
+              isNodeLoading={isNodeLoading}
+              lineageLeafNodes={leafNodes}
+              loadNodeHandler={loadNodeHandler}
+              removeLineageHandler={removeLineageHandler}
+            />
+          </div>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx:318]: We can use `Card` component here

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx [hunk 2]
```diff
@@ -96,15 +96,9 @@ export const getHeaderLabel = (
...
-  reactFlowInstance: ReactFlowInstance,
-  length?: number,
-  forceZoomReset = false
-) => {
+export const onLoad = (reactFlowInstance: ReactFlowInstance) => {
   reactFlowInstance.fitView();
-  if (forceZoomReset || (length && length <= 2)) {
-    reactFlowInstance.zoomTo(zoomValue);
-  }
+  reactFlowInstance.zoomTo(zoomValue);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:101]: From where we are getting this `zoomValue` it should be constant or via function params

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 3]
```diff
@@ -78,13 +99,34 @@ const CustomControls: FC<ControlProps> = ({
...
+      <SVGIcons
+        alt="icon-edit-lineag"
+        className="m--t-xss"
+        icon={isEditMode ? 'icon-edit-lineage' : 'icon-edit-lineage-color'}
+        width="14"
+      />
+    );
+
+    return LoadingStatus(editIcon, loading, status);
+  }, [loading, status, isEditMode]);
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:113]: so here we are getting a button based on the loading state from `LoadingStatus(editIcon, loading, status)` in which editicon is default value there

[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:113]: got it, Thank 🙏🏻

File: openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx:101]: it's from constant!

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 4]
```diff
@@ -126,10 +168,33 @@ const CustomControls: FC<ControlProps> = ({
...
+        <ControlButton
+          className={classNames('h-9 w-9 rounded-full p-x-xss tw-shadow-lg', {
+            'bg-primary': isEditMode,
+            'bg-primary-hover-lite': !isEditMode,
+          })}
+          data-testid="edit-lineage"
+          disabled={!hasEditAccess}
+          title={hasEditAccess ? 'Edit Lineage' : NO_PERMISSION_FOR_ACTION}
+          onClick={onEditLinageClick}>
+          {getLoadingStatus()}
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:195]: updated

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:113]: updated

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx [hunk 5]
```diff
@@ -1037,13 +1002,18 @@ const EntityLineageComponent: FunctionComponent<EntityLineageProp> = ({
             onNodeMouseLeave={onNodeMouseLeave}
             onNodeMouseMove={onNodeMouseMove}
             onNodesChange={onNodesChange}>
-            <CustomControlElements
+            <CustomControlsComponent
+              className="absolute top-1 right-3 bottom-full m-l-md m-t-md"
               deleted={deleted}
+              fitViewParams={{ minZoom: 0.5, maxZoom: 2.5 }}
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx:1008]: moved to constant

File: openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx:318]: updated

**Comment by ShaileshParmar11** [comment]
Hi @chirag-madlani Address all the comments Thanks!

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 6]
```diff
@@ -192,7 +191,7 @@ const CustomControls: FC<ControlProps> = ({
           disabled={!hasEditAccess}
           title={hasEditAccess ? 'Edit Lineage' : NO_PERMISSION_FOR_ACTION}
           onClick={onEditLinageClick}>
-          {getLoadingStatus()}
+          {LoadingStatus(editIcon, loading, status)}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:194]: Sorry for not catching it earlier but function name should be more descriptive & naming should be in camelCase

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx [hunk 7]
```diff
@@ -47,7 +47,11 @@ import ReactFlow, {
   useNodesState,
 } from 'reactflow';
 import { getTableDetails } from '../../axiosAPIs/tableAPI';
-import { ELEMENT_DELETE_STATE } from '../../constants/Lineage.constants';
+import {
+  ELEMENT_DELETE_STATE,
+  maxZoomValue,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx:52]: constants must be in SCREAMING_SNACK_CASE

File: openmetadata-ui/src/main/resources/ui/src/constants/Lineage.constants.ts [hunk 8]
```diff
@@ -4,6 +4,8 @@ import { EntityType } from '../enums/entity.enum';
 
 export const foreignObjectSize = 40;
 export const zoomValue = 1;
+export const minZoomValue = 0.5;
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/Lineage.constants.ts:7]: constants must be in SCREAMING_SNACK_CASE

File: openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx [hunk 9]
```diff
@@ -300,9 +304,9 @@ const LineagePage = () => {
   return (
     <PageContainerV1>
       <PageLayoutV1 className="p-x-lg">
-        <div className="tw-flex tw-flex-col tw-gap-4 h-full">
+        <div className="lineage-page-container">
           <TitleBreadcrumb titleLinks={titleBreadcrumb} />
-          <div className="p-y-sm p-x-lg h-full tw-flex-1 tw-bg-white">
+          <Card className="h-full">
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx:309]: `Card` will accept title prop. Please check if we can use it. If not why we can't

File: openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/lineagePage.style.less [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+  height: 100%;
+
+  .ant-card-body {
+    height: 100%;
+    margin-right: 14px;
+    margin-left: 14px;
+  }
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/lineagePage.style.less:24]: Use size prop of card for margins
Probably className h-full for height

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx (hunk 9 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/LineagePage.tsx:309]: as discussed, with these changes our page styling will be changed and we don't want that 
image

File: openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/lineagePage.style.less (hunk 10 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/lineagePage.style.less]: updated

File: openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/lineagePage.style.less [hunk 11]
```diff
@@ -19,7 +19,7 @@
 
   .ant-card-body {
     height: 100%;
-    margin-right: 14px;
-    margin-left: 14px;
+    // margin-right: 14px;
+    // margin-left: 14px;
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/LineagePage/lineagePage.style.less:23]: Can we remove this?

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**